docs/decisions/0002-remove-use-jwt-cookie-header.rst

```diff
@@ -0,0 +1,52 @@
+2. Remove HTTP_USE_JWT_COOKIE Header
+====================================
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+This ADR explains `why the request header HTTP_USE_JWT_COOKIE`_ was added.
+
+Use of this header has several problems:
+
+* The purpose and need for the ``HTTP_USE_JWT_COOKIE`` header is confusing. It has led to developer confusion when trying to test API calls using JWT cookies, but having the auth fail because they didn't realize this special header was also required.
+* In some cases, the JWT cookies are sent to services, but go unused because of this header. Additional oauth redirects then become required in circumstances where they otherwise wouldn't be needed.
+* Some features have been added, like `JwtRedirectToLoginIfUnauthenticatedMiddleware`_, that can be greatly simplified or possibly removed altogether if the ``HTTP_USE_JWT_COOKIE`` header were retired.
+
+
+Decision
+--------
+
+Replace the `HTTP_USE_JWT_COOKIE` header with forgiving authentication when using JWT cookies. By "forgiving", we mean that JWT authentication would no longer raise exceptions for failed authentication when using JWT cookies, but instead would simply return None.
+
+By returning None from JwtAuthentication, rather than raising an authentication failure, we enable services to move on to other classes, like SessionAuthentication, rather than aborting the authentication process. Failure messages could still be surfaced using `set_custom_metric` for debugging purposes.
+
+Rather than checking for the `HTTP_USE_JWT_COOKIE`, the `JwtAuthCookieMiddleware`_ would always reconstitute the JWT cookie if the parts were available.
+
+Although this would ultimately be a breaking API change for JwtAuthentication (where enabled), the plan would be for the backward incompatibility to be encapsulated and fixed in ``edx-drf-extensions``.
```

Comment:
1. I'm removing this comment, because I don't find it useful. This change is encapsulated to this library, so this is not a backward incompatible change.
```suggestion
```
2. I did find one bit of cleanup in an edx-platform test that can come when we remove the temporary toggle and this header is no longer used. See https://github.com/openedx/edx-platform/blob/f0a9ef21613a780e4c7975de47375fe7f21e3710/openedx/core/djangoapps/user_authn/tests/test_cookies.py#L78.
3. We need a minor version bump and changelog entry for this change.

Reply:
Are you saying it's backwards compatible because we can roll back the library version of because of some other reason?

Reply:
I mean that upgrading to this version of the library _shouldn't_ break anything, including when the new setting is toggled on.

Reply:
@feanil: Can we commit my suggestion (to delete text) above, or do you still have a question about this?